.github/workflows/scripts/comments.mjs

```diff
@@ -90,53 +89,106 @@ export async function commentDiff(
 /**
  * Generates comment for stats
  *
+ * @param {FileSize[]} fileSizeData
+ * @param {FileSize[]} modulesData
  * @param {GithubActionContext} githubActionContext
  * @param {number} issueNumber
  * @param {DiffComment} statComment
  */
 export async function commentStats(
+  fileSizeData,
+  modulesData,
   githubActionContext,
   issueNumber,
-  { path, titleText, markerText }
+  { titleText, markerText }
 ) {
   const reviewAppURL = getReviewAppUrl(issueNumber)
+  let bodyText = ''
+  let fileSizeText = ''
+  let modulesText = ''
+
+  if (!fileSizeData.length && !modulesData.length) {
+    bodyText = 'No changes to any file sizes!'
+  } else {
+    if (fileSizeData.length) {
+      // File sizes
+      const fileSizeTitle = '### File sizes'
+      const fileSizeRows = (await getReadableFileSizes(fileSizeData)).map(
+        (fileSize) => Object.values(fileSize)
+      )
 
-  const distPath = join(path, 'dist')
-  const packagePath = join(path, 'packages/govuk-frontend/dist/govuk')
-
-  // File sizes
-  const fileSizeTitle = '### File sizes'
-  const fileSizeRows = [
-    ...(await getFileSizes(join(distPath, '**/*.{css,js,mjs}'))),
-    ...(await getFileSizes(join(packagePath, '*.{css,js,mjs}')))
-  ]
-
-  const fileSizeHeaders = ['File', 'Size']
-  const fileSizeTable = renderTable(fileSizeHeaders, fileSizeRows)
-  const fileSizeText = [fileSizeTitle, fileSizeTable].join('\n')
-
-  // Module sizes
-  const modulesTitle = '### Modules'
-  const modulesRows = (await Promise.all(modulePaths.map(getStats))).map(
-    ([modulePath, moduleSize]) => {
-      const { base, dir, name } = parse(modulePath)
+      const fileSizeHeaders = ['File', 'Size', 'Percentage change']
+      const fileSizeTable = renderTable(fileSizeHeaders, fileSizeRows)
+      fileSizeText = [fileSizeTitle, fileSizeTable].join('\n')
+    } else {
+      fileSizeText = 'No changes to asset file sizes.'
+    }
 
-      const statsPath = `docs/stats/${dir}/${name}.html`
-      const statsURL = new URL(statsPath, reviewAppURL)
+    if (modulesData.length) {
+      // Module sizes
+      const modulesTitle = '### Modules'
+      // Create an empty array to pass to the reducer so that jsdoc doesn't thing
+      // that the default accumulator value is `never[]`
+      /** @type {Array<Array<string|number>>} */
+      const modulesRowsEmpty = []
+      const modulesRows = (await getReadableFileSizes(modulesData)).reduce(
```

Comment:
I _think_ I understand what's going on here (we're merging the bundled and unbundled stats, right?) but it's not immediately clear.

I wonder if there's a clearer approach using [`Object.groupBy`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy) to first group the rows by the file path, then map the resulting array to build the table data?

Alternatively, I'd suggest adding a comment to explain what's going on.

Reply:
Happy to pair on this if you want to look at refactoring it.

Reply:
`Object.groupBy` looks to be exactly what I want! I really don't like that `reduce` so I'd rather try to merge the data together in a neater way. I'm gonna have a go.